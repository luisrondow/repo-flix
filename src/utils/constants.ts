export const TECHS = ['Typescript', 'Javascript', 'Vue', 'Go', 'CSS', 'Node'] as const

export const BOOKMARKS_KEY = 'repoflix:bookmarks' as const
export const SELECTED_TECHS_KEY = 'repoflix:selected-techs' as const
