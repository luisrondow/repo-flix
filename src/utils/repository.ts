export type Repository = {
  id: string
  name: string
  url: string
  image: string
}
