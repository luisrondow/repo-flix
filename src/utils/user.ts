export type User = {
  uid: string
  email: string
  accessToken: string
  username: string
}
