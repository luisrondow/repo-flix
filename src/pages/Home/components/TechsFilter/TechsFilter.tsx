import ChipsButton from '../../../../components/ChipsButton'
import Text from '../../../../components/Text'
import { useSelectedTechsContext } from '../../../../hooks/useSelectedTechsContext'
import { TECHS } from '../../../../utils/constants'

const TechsFilter = () => {
  const { selectedTechs, actions } = useSelectedTechsContext()

  const handleTechClick = (techName: string) => {
    if (selectedTechs?.includes(techName)) {
      actions?.removeTech(techName)
    } else {
      actions?.addTech(techName)
    }
  }

  return (
    <div className="flex w-full flex-col px-4 sm:px-16">
      <Text as="h2" className="mb-2">
        Toggle topics to show
      </Text>
      <div className="flex w-full flex-row flex-wrap justify-between space-x-2 sm:justify-start sm:space-x-4">
        {TECHS.map((techName) => (
          <ChipsButton
            key={techName}
            isSelected={selectedTechs?.includes(techName)}
            onClick={() => handleTechClick(techName)}
          >
            {techName}
          </ChipsButton>
        ))}
      </div>
    </div>
  )
}

export default TechsFilter
