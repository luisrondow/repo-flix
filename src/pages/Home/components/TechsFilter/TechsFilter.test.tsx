import { render } from '@testing-library/react'
import TechsFilter from './TechsFilter'
import { TECHS } from '../../../../utils/constants'

describe('TechsFilter', () => {
  it('should render correctly', () => {
    const { container, getByText, getAllByTestId } = render(<TechsFilter />)

    expect(getByText('Toggle topics to show')).toBeInTheDocument()
    expect(getAllByTestId('chips-button')).toHaveLength(TECHS.length)

    TECHS.forEach((techName) => {
      expect(getByText(techName)).toBeInTheDocument()
    })

    expect(container).toMatchSnapshot()
  })
})
