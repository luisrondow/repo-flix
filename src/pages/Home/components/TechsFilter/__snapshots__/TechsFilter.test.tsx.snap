// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechsFilter should render correctly 1`] = `
<div>
  <div
    class="flex w-full flex-col px-4 sm:px-16"
  >
    <h2
      class="mb-2 text-[#e0def4] font-roboto font-bold sm:text-xl"
    >
      Toggle topics to show
    </h2>
    <div
      class="flex w-full flex-row flex-wrap justify-between space-x-2 sm:justify-start sm:space-x-4"
    >
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          Typescript
        </div>
      </div>
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          Javascript
        </div>
      </div>
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          Vue
        </div>
      </div>
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          Go
        </div>
      </div>
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          CSS
        </div>
      </div>
      <div
        class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
        data-testid="chips-button"
      >
        <div
          class="max-w-full flex-initial text-lg font-bold leading-none"
        >
          Node
        </div>
      </div>
    </div>
  </div>
</div>
`;
